users/src/main/java/io/crate/operation/auth/ClientCertAuth.java

```diff
@@ -46,51 +44,16 @@
     @Override
     public User authenticate(String userName, ConnectionProperties connProperties) {
         Certificate clientCert = connProperties.clientCert();
-        if (connProperties.hasSSL() && clientCert != null) {
-            if (clientCert instanceof X509Certificate) {
-                String CN = extractCN(((X509Certificate )clientCert).getSubjectX500Principal().getName());
-                User user = lookupUser(userName, connProperties, CN);
-                if (user != null) return user;
-            }
-        }
-        throw new RuntimeException("Client certificate authentication failed for user \"" + userName + "\"");
-    }
-
-    @Nullable
-    private User lookupUser(String userName, ConnectionProperties connProperties, String CN) {
-        // userName is optional in HTTP;
-        if (connProperties.protocol() == Protocol.HTTP) {
-            if (userName == null || userName.equals("")) {
-                userName = CN;
-            }
-        }
-        if (userName.equals(CN)) {
-            User user = userLookup.findUser(userName);
-            if (user != null) {
-                return user;
-            }
-        }
-        return null;
-    }
-
-    private static String extractCN(String subjectDN) {
-        /*
-         * Get commonName using LdapName API
-         * The DN of X509 certificates are in rfc2253 format. Ldap uses the same format.
-         *
-         * Doesn't use X500Name because it's internal API
-         */
-        try {
-            LdapName ldapName = new LdapName(subjectDN);
-            for (Rdn rdn : ldapName.getRdns()) {
-                if ("CN".equalsIgnoreCase(rdn.getType())) {
-                    return rdn.getValue().toString();
+        if (clientCert != null) {
+            String CN = SSL.extractCN(clientCert);
+            if (Objects.equals(userName, CN)) {
```

Comment:
i think it would be good to throw a special exception here, which would be convinient for the user to see what's actually wrong. we do not leak anything when doing this, afaiks.

pls also add a test for this case